docs/python-api.rst

```diff
@@ -126,9 +126,41 @@ using the :ref:`Tables API <sec_tables_api>`.
 
 .. _sec_tables_api:
 
-******
+****************************
+Tables and Table Collections
+****************************
+
+The information required to construct a tree sequence is stored in a collection
+of *tables*, each defining a different aspect of the structure of a tree
+sequence. These tables are described individually in :ref:`the next section
+<sec_tables_docs>`. However, these are interrelated, and so many operations work
+on the entire collection of tables, known as a ``TableCollection``.
+The :class:`TableCollection` and :class:`TreeSequence` classes are
+deeply related. A ``TreeSequence`` instance is based on the information
+encoded in a ``TableCollection``. Tree sequences are **immutable**, and
+provide methods for obtaining trees from the sequence. A ``TableCollection``
+is **mutable**, and does not have any methods for obtaining trees.
+The ``TableCollection`` class thus allows dynamic creation and modification of
+tree sequences.
+
++++++++++++++++++++++++++++++
+The ``TableCollection`` class
++++++++++++++++++++++++++++++
+
+Many of the ``TreeSequence`` methods that return a modified tree sequence
+are in fact wrappers around a corresponding ``TableCollection`` method
+that modifies a copy of the origin tree sequence's table collection.
+
+.. autoclass:: tskit.TableCollection(sequence_length=0)
+    :autosummary:
+    :members:
+
+
+.. _sec_tables_docs:
```

Comment:
Can we change the lable to ``sec_tables_api_tables`` for consistency please? I've tried to make it reasonably consistent elsewhere (but not entirely successfully). The current label doesn't really fit with any conventions though.